### ISSUE TITLE
Add documentation for BackendLBPolicy

### ISSUE THREAD

**Issue by gcs278** [kubernetes-sigs/gateway-api#3201]
**What would you like to be added**:

Similar to https://gateway-api.sigs.k8s.io/api-types/backendtlspolicy/, we should have a dedicated documentation page for [BackendLBPolicy](https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1alpha2.BackendLBPolicy), which was released in v1.1. 

The documentation should describe the goals of `BackendLBPolicy`, how to use the policy, and descriptions of the fields.

**Why this is needed**:

Though the current usage of `BackendLBPolicy` is documented well in [GEP-1619](https://gateway-api.sigs.k8s.io/geps/gep-1619/), we should have a user-facing doc to be consistent with `BackendTLSPolicy`, as we don't expect end-users to dig through GEPs to better understand how a policy works.

**Comment by gcs278** [comment]
@shaneutt @robscott Do you agree with the documentation need here? I noticed `BackendTLSPolicy` has a dedicated page, so it made sense to me to dedicate one to `BackendLBPolicy`. What do you think the priority of this issue is?

**Comment by shaneutt** [comment]
@gcs278 I agree that since this is out as experimental we should document it similarity as you proposed.

/triage accept

As for priority it can be hard to say. I would recommend you put an agenda item to talk about this our next sync.

**Comment by youngnick** [comment]
Big +1 from me, I agree this should have a user-facing doc that we can keep up to date and can be the source of truth for how this works.

**Comment by amitmaurya07** [comment]
Hi, I would like to work on this issue. Could you please assign this issue to me?

**Comment by robscott** [comment]
Sure thing, thanks for volunteering @amitmaurya07!

/assign @amitmaurya07

**Comment by gcs278** [comment]
@amitmaurya07 Thanks! Feel free to reach out on here, or on the [#sig-network-gateway-api](https://kubernetes.slack.com/archives/CR0H13KGA) slack channel if you have any questions.

**Comment by amitmaurya07** [comment]
Sure, Thanks @gcs278 @robscott
